aioauth/server.py

```diff
@@ -313,6 +313,17 @@ async def token(request: fastapi.Request) -> fastapi.Response:
         )
 
         client = await grant_type.validate_request(request)
+        # If a client has been issued a client_secret, it MUST be used with
+        # password grant type.
+        if (
```

Comment:
This is the important part of the change. I'm open to doing this another way, but it didn't seem like the grant type had the info it needed to make this assertion, so I added it here.